packages/charts/react-charting/src/components/ResponsiveContainer/ResponsiveContainer.tsx

```diff
@@ -0,0 +1,73 @@
+import * as React from 'react';
+import { classNamesFunction, getWindow } from '@fluentui/react';
+import {
+  IResponsiveChildProps,
+  IResponsiveContainerProps,
+  IResponsiveContainerStyles,
+} from './ResponsiveContainer.types';
+import { getStyles } from './ResponsiveContainer.styles';
+
+const getClassNames = classNamesFunction<{}, IResponsiveContainerStyles>();
+
+/**
+ * Responsive Container component
+ * {@docCategory ResponsiveContainer}
+ */
+export const ResponsiveContainer: React.FC<IResponsiveContainerProps> = props => {
+  const containerRef = React.useRef<HTMLDivElement>(null);
+  const onResizeRef = React.useRef<IResponsiveContainerProps['onResize']>();
+  const classNames = React.useMemo(() => getClassNames(getStyles()), []);
+
+  const [size, setSize] = React.useState<{ containerWidth?: number; containerHeight?: number }>({});
+
+  onResizeRef.current = props.onResize;
+
+  React.useEffect(() => {
+    const _window = getWindow(containerRef.current) as (Window & typeof globalThis) | undefined;
+    let animationFrameId: number | undefined;
+    let resizeObserver: ResizeObserver | undefined;
+
+    const resizeCallback = (entries: ResizeObserverEntry[]) => {
+      const { width: containerWidth, height: containerHeight } = entries[0].contentRect;
```

Comment:
>0 [](http://example.com/codeflow?start=73&length=1)

is 0 guaranteed to be present

Reply:
Yes, the callback won't be triggered without an entry. See this: https://stackoverflow.com/a/69690616